sunpy/map/sources/iris.py

```diff
@@ -42,8 +42,8 @@ def __init__(self, data, header, **kwargs):
         super().__init__(data, header, **kwargs)
 
         self.meta['detector'] = "SJI"
-        self.meta['waveunit'] = "Angstrom"
-        self.meta['wavelnth'] = header['twave1']
+        self.meta['waveunit'] = header.get('waveunit', "Angstrom")
+        self.meta['wavelnth'] = header.get('wavelnth', 'twave1')
```

Comment:
This is closer to correct, but could still end up with `None` in there. Although it would error currently.
```suggestion
        self.meta['wavelnth'] = header.get('wavelnth', header.get('twave1'))
```